syncstorage/storage/sql/__init__.py

```diff
@@ -493,7 +493,11 @@ def create_batch(self, session, userid, collection):
             "userid": userid,
             "collection": collectionid
         }
-        session.query("CREATE_BATCH", params)
+        try:
+            session.query("CREATE_BATCH", params)
+        except IntegrityError:
+            # The user tried to create two batches with the same timestamp.
```

Comment:
no; the primary key on this table is (batchid, userid) so it'll only trigger for duplicates of this tuple.

(This change isn't critical to the issue at hand, but was motivated by the new testcase.  It was possible for a user to trigger a `500 Server Error` by creating a batch and then quickly creating another without closing the first.  Now that will produce a `409 Conflict` which is more appropriate)